### PR TITLE
feat: implement GatherND operator

### LINKED ISSUE #9: Implement ScatterND ONNX Operation
## Feature description
Add support for the ONNX ScatterND operation in the `burn-import` crate by implementing the appropriate conversion pathway. This will allow ONNX models using ScatterND to run in Burn.

## Feature motivation
ScatterND is a flexible tensor indexing op widely used in contemporary ONNX models. Adding this op will improve compatibility and completeness of model imports.

## (Optional) Suggest a Solution
- Use GatherND and other multidimensional ops as a reference for implementation techniques. 
- Develop thorough tests covering edge cases.

Further references:
- [SUPPORTED-ONNX-OPS. md](https://github.com/tracel-ai/burn/blob/main/crates/burn-import/SUPPORTED-ONNX-OPS.md)
- [Contributor book guide](http://localhost:3011/guides/onnx-to-burn-conversion-tool.html)
- Master tracking issue tracel-ai/burn#1714

Related context: https://github.com/tracel-ai/burn/issues/1714

### LINKED ISSUE #97: Unsupported operation: GatherND
**Describe the bug**
Importing ONNX models that use the `GatherND` operation fails with:

```
Unsupported node type for inputs: GatherND(GatherNDNode { name: "gathernd1", inputs: [...], outputs: [...] })
```

**To Reproduce**
1. Download a model that uses GatherND, such as the meikiocr detection model:
   ```bash
   curl -L -o detect_tiny.onnx "https://huggingface.co/rtr46/meiki.text.detect.v0/resolve/main/meiki.text.detect.tiny.v0.onnx"
   ```
2. Run burn-onnx:
   ```bash
   cargo run -p burn-onnx -- detect_tiny.onnx ./out
   ```

**Expected behavior**
Either support GatherND or emit a clear error message indicating the operation is not supported.

**ONNX file**
https://huggingface.co/rtr46/meiki.text.detect.v0/resolve/main/meiki.text.detect.tiny.v0.onnx

**Additional context**
Found while testing fix for #91. After fixing the MatMul 1D weight panic, the model import progresses further but fails on GatherND.

### PR DESCRIPTION
**Pull request by antimora**
## Summary
- Implements the ONNX `GatherND` operator (opset 11+) with full `batch_dims` support and negative index handling
- Adds onnx-ir node processor with type inference, burn-onnx codegen, and 5 integration tests covering spec examples
- Moves GatherND from unsupported placeholder to fully supported operator

## Test plan
- [x] 11 unit tests in onnx-ir (type inference, config extraction, validation)
- [x] 5 insta snapshot tests in burn-onnx (codegen for various ranks, batch_dims, dtypes)
- [x] 5 integration tests in onnx-tests (spec examples 1-3, 5; negative indices)
- [x] All existing tests pass (681 onnx-ir, 467 burn-onnx, 438 onnx-tests)

Closes #97
Closes #9

### CODE REVIEW

File: crates/onnx-ir/src/node/gathernd.rs [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+        }
+
+        Ok(())
+    }
+
+    fn extract_config(&self, node: &RawNode, _opset: usize) -> Result<Self::Config, ProcessError> {
+        let mut batch_dims: i64 = 0;
+        for (key, value) in node.attrs.iter() {
+            if key.as_str() == "batch_dims" {
+                batch_dims = value.clone().into_i64();
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/gathernd.rs]: Fixed. Added a match statement in `extract_config` that rejects unknown attributes with `ProcessError::InvalidAttribute`, consistent with the validation already in `infer_types`.

File: crates/burn-onnx/src/burn/node/gathernd.rs [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+                    }
+                }
+
+                // Compute output shape: data_dims[:b] + indices_dims[b:q-1] + data_dims[b+k:]
+                let mut output_shape = [0usize; #output_rank_lit];
+                let mut si = 0;
+                for i in 0..b {
+                    output_shape[si] = data_dims[i];
+                    si += 1;
+                }
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-onnx/src/burn/node/gathernd.rs]: These branches are trivially eliminated by the compiler when `b = 0` is a constant. The codegen would need to emit two entirely different code paths (one for batch_dims=0, one for batch_dims>0), adding complexity to the codegen without a measurable runtime benefit. Keeping the general code path that works for both cases.

File: crates/burn-onnx/src/burn/node/gathernd.rs [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+                for i in b + k..r {
+                    output_shape[si] = data_dims[i];
+                    si += 1;
+                }
+                output_flat.reshape(output_shape)
+            };
+            output
+        }
+        ");
+    }
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-onnx/src/burn/node/gathernd.rs:658]: Fixed. Added `test_gathernd_scalar_output` snapshot test covering the output_rank=0 (scalar) codegen path.

File: crates/burn-onnx/src/burn/node/gathernd.rs [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+                    output_shape[si] = indices_dims[i];
+                    si += 1;
+                }
+                for i in b + k..r {
+                    output_shape[si] = data_dims[i];
+                    si += 1;
+                }
+
+                output_flat.reshape(output_shape)
+            };
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-onnx/src/burn/node/gathernd.rs]: Fixed. The scalar output path now generates dedicated code that uses `into_scalar()` instead of trying to reshape with a zero-sized array. The tensor and scalar paths are now separate branches in the codegen.

File: crates/onnx-tests/tests/gathernd/gathernd.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+import numpy as np
+import onnx
+from onnx import TensorProto, helper
+from onnx.reference import ReferenceEvaluator
+
+
+def compute_gathernd_output_shape(data_shape, indices_shape, batch_dims=0):
+    """Compute output shape per the ONNX GatherND spec."""
+    r = len(data_shape)
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-tests/tests/gathernd/gathernd.py]: Fixed. Removed the unused `r` variable from `compute_gathernd_output_shape`.
